.github/workflows/examples-java-push.yml

```diff
@@ -0,0 +1,43 @@
+name: Build `examples/java` pushes
+
+on:
+  push:
```

Comment:
could this be one yml file instead of two (I think it's all the same?), ie add this `push` block to the `on` block in examples-java-pr? https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-using-multiple-events-with-activity-types-or-configuration

Reply:
Although right now they're the same, the plan is that they're going to be different as we'll be doing things like publishing on `main` or running SonarQube linting on branches, so I thought I'd deduplicate them for now